process/outportBlockCaster.go

```diff
@@ -0,0 +1,186 @@
+package process
+
+import (
+	"fmt"
+	"math/big"
+
+	coreData "github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/block"
+	"github.com/multiversx/mx-chain-core-go/data/outport"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/multiversx/mx-chain-ws-connector-template-go/data"
+)
+
+var (
+	protoMarshaller = &marshal.GogoProtoMarshalizer{}
+	bigIntCaster    coreData.BigIntCaster
+)
+
+type (
+	headerV1Caster struct {
+		outportBlock *outport.OutportBlock
+	}
+	headerV2Caster struct {
+		outportBlock *outport.OutportBlock
+	}
+	metaBlockCaster struct {
+		outportBlock *outport.OutportBlock
```

Comment:
another way to implement here if to just use 3 functions for each converter, and avoid having these 3 structs, since the converters will only use the provided outportblock

Reply:
or we can have gogo proto marshaller set as parameter to a converter component (and maybe even a separate component for standard proto marshaller that implements our `Marshaller` interface), which will have 3 methods (like `HeaderV1Converter`, `HeaderV1Converter`, `MetaBlockConverter`)

Reply:
refactored the component.